apps/api/src/domain/common/events/sync-in-memory-event-publisher.ts

```diff
@@ -48,6 +48,19 @@ export class SyncInMemoryEventPublisher implements ICoscradEventPublisher {
             this.eventTypeToConsumers.set(eventType, []);
         }
 
-        this.eventTypeToConsumers.get(eventType).push(eventConsumer);
+        /**
+         * We want registration to be idempotent in case this module is somehow
```

Comment:
Is this done in the other implementation as well?

Reply:
It's difficult to see how this can easily be done using the rxjs implementation. I think it's best to deprecate \ remove this because it is much more opaque than the other implementation. Eventually, we'll move to a proper messaging queue, so let's not overthink this.

Reply:
It's difficult to see how this can easily be done using the rxjs implementation. I think it's best to deprecate \ remove this because it is much more opaque than the other implementation. Eventually, we'll move to a proper messaging queue, so let's not overthink this.